compiler/src/iree/compiler/Codegen/Common/Transforms.cpp

```diff
@@ -728,4 +735,99 @@ void populateSwapExtractWithCollapsePattern(RewritePatternSet &patterns) {
   patterns.add<SwapCollapseShapeWithSlicePattern>(patterns.getContext());
 }
 
+namespace {
+
+struct RemoveOptimizationBarrier final
```

Comment:
Ouch, this is a bad pattern. We should look into removing it from that BlockDynamicDimensions pass in the future.